File: cpp/34.cpp
// https://leetcode.com/problems/search-for-a-range/
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        auto itpair = std::equal_range(std::begin(nums), std::end(nums), target);
        if (itpair.first == std::end(nums) || *itpair.first != target) return {-1, -1};
        return {std::distance(std::begin(nums), itpair.first),
                std::distance(std::begin(nums), std::prev(itpair.second)) };
    }
};

File: golang/34.go
package main

func searchRange(nums []int, target int) []int {
	idx1 := sort.Search(len(nums), func(i int) bool { return nums[i] >= target })
	idx2 := sort.Search(len(nums), func(i int) bool { return nums[i] > target })

	fmt.Println(idx1, idx2)
	l := len(nums)

	if idx1 == l || nums[idx1] != target {
		return []int{-1, -1}
	}

	return []int{idx1, idx2 - 1}
}
